src/xmldoc.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+Copyright 2015 University of Auckland
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <string>
+#include <cstring>
+#include <vector>
+#include <libxml/parser.h>
+#include <libxml/tree.h>
+#include <libxml/xmlerror.h>
+
+#include "xmldoc.h"
+#include "xmlnode.h"
+
+
+namespace libcellml {
+
+/**
+ * @brief Callback for errors from the libxml2 context parser.
+ *
+ * Structured callback @c xmlStructuredErrorFunc for errors
+ * from the libxml2 context parser used to parse this document.
+ *
+ * @param userData Private data type used to store the libxml context.
+ *
+ * @param error The @c xmlErrorPtr to the error raised by libxml.
+ */
+void structuredErrorCallback(void *userData, xmlErrorPtr error)
+{
+  std::string errorString = std::string(error->message);
+  errorString.replace(errorString.end()-1, errorString.end(), ".");
```

Comment:
Ok, that's fair enough then, but... I would still check that the last character is indeed `\n`, in case libxml2 was ever to change the formatting of their errors (better be safe than sorry...).